src/main/scala/devices/i2c/I2CPeriphery.scala

```diff
@@ -19,15 +19,21 @@ trait HasPeripheryI2C extends HasTopLevelNetworks {
   }
 }
 
-trait HasPeripheryI2CBundle extends HasTopLevelNetworksBundle{
-  val outer: HasPeripheryI2C
-  val i2cs = Vec(outer.i2cParams.size, new I2CPort)
+trait HasPeripheryI2CBundle {
+  val i2cs: Vec[I2CPort]
+
+  def toGPIOPins(dummy: Int = 1): Seq[I2CGPIOPort] = i2cs.map { i =>
```

Comment:
Members of chisel bundles that don't take parameters are automatically converted into wires. So this is the standard gross hack to avoid that from happening before you want it to. Happy to replace it with a parameter that is actually useful.